src/validation_workflow/validation.py

```diff
@@ -7,15 +7,44 @@
 
 
 from abc import ABC, abstractmethod
+from typing import Any
 
-from system.temporary_directory import TemporaryDirectory
+from validation_workflow.validation_args import ValidationArgs
 
 
 class Validation(ABC):
-    base_url = "https://artifacts.opensearch.org/releases/bundle/"
-    tmp_dir = TemporaryDirectory()
 
-    @classmethod
+    def __init__(self, args: ValidationArgs) -> None:
+        super().__init__()
+        self.args = args
+
+    def run(self) -> Any:
+        try:
+            if self.download_artifacts(self.args):
+                if self.installation(self.args):
+                    if self.start_cluster(self.args):
+                        if self.validation(self.args):
+                            if self.cleanup(self.args):
+                                return True
```

Comment:
Can we use `&` here instead of nested `if`?

Reply:
or just `return self.download_artifacts(self.args) && self.installation(self.args) & ...`

Reply:
Thanks a lot for the feedback, @zelinh .. indeed, we want to return True only when all of those sub processes returned True.. we can consolidate them with && indeed. Let us update it in the next commit.